src/parser.ts

```diff
@@ -305,17 +306,63 @@ function parseSuperTypes(
   options: Options,
   processed: Processed,
   usedNames: UsedNames
-): TNamedInterface[] {
+): TSuperType[] {
   // Type assertion needed because of dereferencing step
   // TODO: Type it upstream
   const superTypes = schema.extends as SchemaSchema | SchemaSchema[] | undefined
   if (!superTypes) {
     return []
   }
   if (Array.isArray(superTypes)) {
-    return superTypes.map(_ => newNamedInterface(_, options, _, processed, usedNames))
+    return superTypes.map(_ => getOrCreateSuperType(_, options, _, processed, usedNames))
+  }
+  return [getOrCreateSuperType(superTypes, options, superTypes, processed, usedNames)]
+}
+
+function isSuperType(type: any): type is TSuperType {
+  return type.type === 'CUSTOM_TYPE' || type.type === 'INTERFACE'
+}
+
+function getOrCreateSuperType(
```

Comment:
Nit: please rename this for consistency
```suggestion
function newSupertype(
```